engine/table/src/main/java/io/deephaven/engine/table/impl/by/HashedRunFinder.java

```diff
@@ -92,8 +89,14 @@ public static boolean findRunsHashed(
                     overflowPointer += 2;
                     break;
                 } else {
-                    // linear probe
-                    hashSlot = (hashSlot + 1) & context.tableMask;
+                    if (probe == 0) {
+                        // double hashing
+                        probe = 1 + (outputPosition % (context.tableSize - 2));
```

Comment:
Are we sure this probe sequence can never result in an infinite loop?

Reply:
Yes, similar to Trove we force `tableSize` to be a prime number and our `h2()` function returns a number in the range `1 <= n < tableSize` which guarantees the second hash has no common factors other than 1 (i.e. relatively prime).

I was initially worried about small desired capacities (0,1,2) but have verified that `PrimeFinder#nextPrime()` returns 3 as the smallest possible prime (i.e. when provided `0 <= x <= 3`). so the small cases are covered properly as well.